### PR TITLE
Implement plugin versions

### LINKED ISSUE #178: Feature Request: version support in plugin module
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.10
- Ruby: built in
- Distribution: centos 7
- Module version: 6 

We need to install a specific version of grafana plugins. It would be great if we could specify this from the puppet provider

so for instance
 ```puppet
grafana_plugin{ 'grafana-clock-panel':
	ensure => "1.0.3";
}
```

### PR DESCRIPTION
**Pull request by TuningYourCode**
#### Pull Request (PR) description
This PR provides a package provider to manage grafana plugins as the current `grafana_plugin` resource does not offer all functions like installation of specific plugin versions or keeping plugins up to date with `ensure => latest`.

This functionality is now given by using
```
package { 'grafana-image-renderer':
  ensure => 'latest',
  provider => 'grafana',
}
```
or
```
package { 'grafana-image-renderer':
  ensure => '2.0.0',
  provider => 'grafana',
}
```

#### This Pull Request (PR) fixes the following issues
Fixes #178

### CODE REVIEW

**Comment by root-expert** [comment]
Is this supposed to be a replacement for `grafana_plugin` ? If yes, maybe we should add a deprecation notice in the docs for user to gradually switch to the new package provider?

BTW master should be fixed now if you want to rebase @TuningYourCode :smile:

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -174,4 +174,30 @@ class { 'grafana':
...
+
+    describe package('grafana-image-renderer') do
+      it { is_expected.to be_installed }
+    end
+
+    describe service('grafana-server') do
+      it { is_expected.to be_enabled }
+      it { is_expected.to be_running }
+    end
+  end
```
Comment (root-expert) [inline spec/acceptance/class_spec.rb:202]: Maybe we could move this to `grafana_plugin_spec.rb`?

**Comment by bastelfreak** [comment]
@TuningYourCode can you please rebase against master to get rid of the merge commits and run `bundle exec rake rubocop:auto_correct`? That should fix the rubocop complaints.

**Review by bastelfreak — APPROVED** [review]

**Comment by SimonHoenscheid** [comment]
@TuningYourCode Can you please rebase and fix the conflicts?

**Pull request closed without merging**